src/tempcharset.h

```diff
@@ -0,0 +1,54 @@
+static struct
```

Comment:
Here is the point of confusion.

This line of code combined with line 31 is doing two different things. First it is defining a `struct` and second it is making a file local variable from that struct. `static` means "global to this file only, invisible otherwise" with the side effect of initializing the value to 0 (aka NULL in this case since it is a pointer).

Remove the static and the variable and give this struct a proper name like `struct CharseWork`. Then you can declare a `static` instance of it in C. Any Rust code that needs the variable will then need to access it via an accessor function like get_temp_charset_work(). It is fine to assign to the pointer. Once we move the rest of the code using the variable to Rust we can switch over to using Rust's `lazy_static` support.